homeassistant/components/verisure/alarm_control_panel.py

```diff
@@ -31,51 +34,53 @@ def setup_platform(
     discovery_info: dict[str, Any] | None = None,
 ) -> None:
     """Set up the Verisure platform."""
+    coordinator = hass.data[DOMAIN]
     alarms = []
-    if int(hub.config.get(CONF_ALARM, 1)):
-        hub.update_overview()
-        alarms.append(VerisureAlarm())
+    if int(coordinator.config.get(CONF_ALARM, 1)):
+        alarms.append(VerisureAlarm(coordinator))
     add_entities(alarms)
 
 
-def set_arm_state(state: str, code: str | None = None) -> None:
-    """Send set arm state command."""
-    transaction_id = hub.session.set_arm_state(code, state)[
-        "armStateChangeTransactionId"
-    ]
-    LOGGER.info("verisure set arm state %s", state)
-    transaction = {}
-    while "result" not in transaction:
-        sleep(0.5)
-        transaction = hub.session.get_arm_state_transaction(transaction_id)
-    hub.update_overview()
-
-
-class VerisureAlarm(AlarmControlPanelEntity):
+class VerisureAlarm(CoordinatorEntity, AlarmControlPanelEntity):
     """Representation of a Verisure alarm status."""
 
-    def __init__(self):
+    coordinator: VerisureDataUpdateCoordinator
+
+    def __init__(self, coordinator: VerisureDataUpdateCoordinator) -> None:
         """Initialize the Verisure alarm panel."""
+        super().__init__(coordinator)
         self._state = None
-        self._digits = hub.config.get(CONF_CODE_DIGITS)
-        self._changed_by = None
 
     @property
     def name(self) -> str:
         """Return the name of the device."""
-        giid = hub.config.get(CONF_GIID)
+        giid = self.coordinator.config.get(CONF_GIID)
         if giid is not None:
-            aliass = {i["giid"]: i["alias"] for i in hub.session.installations}
+            aliass = {
+                i["giid"]: i["alias"] for i in self.coordinator.session.installations
+            }
             if giid in aliass:
                 return "{} alarm".format(aliass[giid])
 
             LOGGER.error("Verisure installation giid not found: %s", giid)
```

Comment:
Side note: We don't want logging in entity state properties. The name calculation should probably move to a coordinator update callback which saves the name in an instance attribute.

Reply:
Honestly, I'm not sure why this is all even here... I skipped a lot of things to keep the PR minimal and continue cleanup in future PRs.